integrations/gemini/actions/response/pipeline/02_generateOutput/main.ts

```diff
@@ -23,12 +24,24 @@ const systemPromptTemplate = await Deno.readTextFile(
 );
 
 export const generateOutput: EventPipelineProcessor = async function () {
-  // If there's nothing to focus on, skip this step.
-  if (
-    !this.data.input.some((event) =>
+  const trace = createTrace(
```

Comment:
ah, I was thinking more...

`this.trace.setAttribute("inputEventCount", this.data.input.length)`

and trace on the pipeline context is bound based on pipeline.name

the trace can be automatically cleaned up when the pipeline ends

we should also add similar functionality to each action! the pipeline trace inherits from the action trace!